blsSignatures/blsSignatures.go

```diff
@@ -118,31 +123,36 @@ func verifySignature2(sig Signature, message []byte, publicKey PublicKey, keyVal
 		return false, err
 	}
 
-	engine := bls12381.NewEngine()
-	engine.Reset()
-	engine.AddPair(pointOnCurve, publicKey.key)
-	leftSide := engine.Result()
-	engine.AddPair(sig, engine.G2.One())
-	rightSide := engine.Result()
-	return leftSide.Equal(rightSide), nil
+	leftSide, err := bls12381.Pair([]bls12381.G1Affine{*pointOnCurve}, []bls12381.G2Affine{*publicKey.key})
+	if err != nil {
+		return false, err
+	}
+	_, _, _, g2 := bls12381.Generators()
+	rightSide, err := bls12381.Pair([]bls12381.G1Affine{*sig}, []bls12381.G2Affine{g2})
+	if err != nil {
+		return false, err
+	}
+	return leftSide.Equal(&rightSide), nil
 }
 
 func AggregatePublicKeys(pubKeys []PublicKey) PublicKey {
-	g2 := bls12381.NewG2()
-	ret := g2.Zero()
+	g2 := new(bls12381.G2Affine)
+	g2.X.SetZero()
+	g2.Y.SetZero()
```

Comment:
It looks to me like `new(bls12381.G2Affine)` will already return something with all zeroes, so I don't think these SetZero calls are necessary.

Reply:
Done.